### PR TITLE
[LoRA] Change lora_tokenizers capacity

### PR DESCRIPTION
**Pull request by xyang16**
Currently lora_tokenizers capacity is max_num_seqs. This is fine if max_loras is smaller than max_num_seqs.

But if max_loras is greater than max_num_seqs, lora_tokenizers should have capacity=max_loras, just like active_adapters LoRALRUCache have capacity=max_loras. If not, items in active_adapters are not evicted, but items in lora_tokenizers are evicted. I think this is unnecessarily evicted, causing performance degradation.

In our benchmark, we test max_loras=50, max_num_seqs=32, 50 adapters are invoked evenly distributed. We found this causing 50% performance degradation in P90 latency, comparing to tokenizers are not evicted.

So raising this PR to make lora_tokenizers capacity to be max(max_loras, max_num_seqs).

Another option is to add max_lora_tokenizers, just like max_loras and max_cpu_loras which controls GPU cache and CPU cache. But this might be an overkill.

### CODE REVIEW

**Comment by jeejeelee** [comment]
Thanks for your contribution, I will look at this PR tomorrow or the day after tomorrow

**Review by jeejeelee — CHANGES_REQUESTED** [review]
Thank you very much for your contribution. Could we pass `max_loras` as a keyword argument?  This way the overall changes would be minimal.

**Comment by xyang16** [comment]
> Thanks for your contribution, I will look at this PR tomorrow or the day after tomorrow

@jeejeelee Thanks for reviewing. There's already **tokenizer_config https://github.com/vllm-project/vllm/blob/main/vllm/transformers_utils/tokenizer_group/tokenizer_group.py#L18, you mean add to **tokenizer_config?

**Comment by jeejeelee** [comment]
> > Thanks for your contribution, I will look at this PR tomorrow or the day after tomorrow
> 
> @jeejeelee Thanks for reviewing. There's already **tokenizer_config https://github.com/vllm-project/vllm/blob/main/vllm/transformers_utils/tokenizer_group/tokenizer_group.py#L18, you mean add to **tokenizer_config?

Yeah

**Comment by xyang16** [comment]
> > > Thanks for your contribution, I will look at this PR tomorrow or the day after tomorrow
> > 
> > 
> > @jeejeelee Thanks for reviewing. There's already **tokenizer_config https://github.com/vllm-project/vllm/blob/main/vllm/transformers_utils/tokenizer_group/tokenizer_group.py#L18, you mean add to **tokenizer_config?
> 
> Yeah

OK

**Comment by xyang16** [comment]
@jeejeelee I have updated the code, please review.

File: vllm/transformers_utils/tokenizer_group/tokenizer_group.py [hunk 1]
```diff
@@ -21,8 +21,9 @@ def __init__(self, tokenizer_id: str, enable_lora: bool, max_num_seqs: int,
         self.enable_lora = enable_lora
         self.max_input_length = max_input_length
         self.tokenizer = get_tokenizer(self.tokenizer_id, **tokenizer_config)
+        max_loras = tokenizer_config.get("max_loras", 0)
         self.lora_tokenizers = LRUCache[AnyTokenizer](
-            capacity=max_num_seqs if enable_lora else 0)
+            capacity=max(max_loras, max_num_seqs) if enable_lora else 0)
```
Comment (jeejeelee) [inline vllm/transformers_utils/tokenizer_group/tokenizer_group.py:26]: We also need to add unit tests for this change, thanks.

Comment (xyang16) [reply vllm/transformers_utils/tokenizer_group/tokenizer_group.py:26]: Added unit tests, thanks.

**Review by jeejeelee — APPROVED** [review]
overall LGTM, thanks for your contribution  and patience

**Pull request merged**